OmgvaPOS/src/ReservationManagement/Controller/ReservationsController.cs

```diff
@@ -0,0 +1,88 @@
+﻿using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using OmgvaPOS.Exceptions;
+using OmgvaPOS.ReservationManagement.DTOs;
+using OmgvaPOS.ReservationManagement.Service;
+using static OmgvaPOS.Exceptions.ExceptionMessages;
+
+namespace OmgvaPOS.ReservationManagement.Controller
+{
+    [ApiController]
+    [Route("reservation")]
+    public class ReservationController : ControllerBase
+    {
+        private readonly IReservationService _reservationService;
+
+        public ReservationController(IReservationService reservationService)
+        {
+            _reservationService = reservationService;
+        }
+
+        [HttpPost]
+        [Authorize(Roles = "Admin,Owner,Employee")]
+        [ProducesResponseType<ReservationDto>(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+        public ActionResult<ReservationDto> CreateReservation([FromBody] CreateReservationDto createDto)
+        {
+            var reservation = _reservationService.Create(createDto);
+            return CreatedAtAction(nameof(GetReservation), new { reservationId = reservation.Id }, reservation);
+        }
+
+        [HttpGet("{reservationId}")]
+        [Authorize(Roles = "Admin,Owner,Employee")]
+        [ProducesResponseType<ReservationDto>(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        // TODO: add forbidden once we link reservation to business
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+        public ActionResult<ReservationDto> GetReservation(long reservationId)
+        {
+            var reservation = _reservationService.GetById(reservationId);
+            if (reservation == null)
+                throw new NotFoundException(ReservationNotFoundMessage(reservationId));
```

Comment:
Currently it seems that this line does exactly the same thing as returning "NotFound(ReservationNotFoundMessage(reservationId))".

Reply:
If we use `NotFound(ReservationNotFoundMessage(reservationId))` then the response format is plain text
![image](https://github.com/user-attachments/assets/e7b23cf4-f327-49fe-8a32-73b964f5e9c0)

With thrown exception we will always have a standard json error response that will include the message 
![image](https://github.com/user-attachments/assets/b096c596-49a9-47d9-976f-f48f231bb6ce)

However agree that in controller we should return NotFound instead of _throw_ it, so passing not the message but ErrorResponse object with that message